### PR TITLE
Added mean temperature and light report at predictions.csv

### PR DESCRIPTION
**Pull request by spiousas**
I added an optional report of the mean temperature and light for each 30-second interval. This will work only for wearables containing temperature and light data stored in columns named `temperature` and `light`, respectively (for example, Axivity AX3).

When adding the parameter `--report_light_and_temp` to the `get_sleep` function call, two columns will be added to the _predictions.csv_ file. They contain the mean light and temperature for the interval in the units and scale the wearable provides.

Use:
```
get_sleep sample.cwa --report_light_and_temp
```

Example output:
```
Total sample count : 10973
13it [05:48, 26.80s/it]
Save predictions to outputs/sample/y_pred.npy
Save prediction probs to outputs/sample/pred_prob.npy
Time used 349.3510031700134
predictions_df shape: (40111, 6)
                     time sleep_wake sleep_stage  raw_label  temperature     light
0 2024-06-14 12:07:38.460       wake        wake          0    20.829166  3.448910
1 2024-06-14 12:08:08.460       wake        wake          0    20.456833  3.242949
2 2024-06-14 12:08:38.460       wake        wake          0    20.450001  3.243973
3 2024-06-14 12:09:08.460       wake        wake          0    20.416834  3.237355
4 2024-06-14 12:09:38.460       wake        wake          0    20.299999  3.241414
Predictions saved to: outputs/sample/predictions.csv
                          start                           end      interval_start        interval_end  wear_duration_H  is_longest_block
0 2024-06-14 17:05:38.460000000 2024-06-14 17:14:38.460000000 2024-06-14 12:00:00 2024-06-15 11:59:59        14.375000              True
1 2024-06-16 01:51:08.459999999 2024-06-16 10:55:38.460000000 2024-06-15 12:00:00 2024-06-16 11:59:59        24.000000              True
2 2024-06-16 17:35:08.460000000 2024-06-16 17:39:08.460000000 2024-06-16 12:00:00 2024-06-17 11:59:59        17.516667             False
3 2024-06-17 00:40:08.460000000 2024-06-17 07:51:08.459999999 2024-06-16 12:00:00 2024-06-17 11:59:59        17.516667              True
4 2024-06-17 17:59:08.460000000 2024-06-17 18:02:08.460000000 2024-06-17 12:00:00 2024-06-18 11:59:59        24.000000             False
```

### CODE REVIEW

**Comment by spiousas** [comment]
It might be a good idea to add an error catch in case columns temperature or light are not present in the raw data.

**Comment by angerhang** [comment]
Hey @spiousas thanks for the PR, would you mind sharing the sample test output and the command used like in this PR https://github.com/OxWearables/asleep/pull/52

File: src/asleep/get_sleep.py [hunk 1]
```diff
@@ -148,6 +148,20 @@ def transform_data2model_input(
...
 
+def mean_temp_and_light(data):
+    # Calculates mean temperature and light for each 30s window
+    print("Calculating mean temperature and light for each 30s interval.")
+    
+    data['time'] = pd.to_datetime(data['time'])
+    # Grouping by time
+    grouped = data.groupby(pd.Grouper(key='time', freq='30S', origin=data['time'].min()))
+    # Calculating the mean temperature and light for each group
+    mean_values = grouped[['temperature', 'light']].mean().iloc[:-1]
```
Comment (angerhang) [inline src/asleep/get_sleep.py:159]: Could you include an expiation handler here to exit the file processing when no temperature or light data is available? No all the devices have these information.

**Review by angerhang — CHANGES_REQUESTED** [review]
Just add the error handling.

File: src/asleep/get_sleep.py (hunk 1 above)
Comment (angerhang) [reply src/asleep/get_sleep.py:159]: Also some of the CI testes failed. Would be good if these are also addressed :D 

https://github.com/OxWearables/asleep/actions/runs/9999071709/job/27810403819?pr=53

Comment (angerhang) [reply src/asleep/get_sleep.py:163]: @spiousas the CI test still failed though. The formatting should be easy to fix :D

Comment (spiousas) [reply src/asleep/get_sleep.py:165]: I am sorry, I missed some fixes. Now it should be OK.

**Review by angerhang — APPROVED** [review]
Looks great!

**Pull request merged**